File: stock_temporal_attention_multilang/novelty4_multi_head_attention/attention.cpp
#include <iostream>
#include <vector>
using namespace std;

int main(){
    vector<double> data = {100,102,101,105,110};
    vector<double> w;
    double sum = 0;
    for(int i=0;i<data.size();i++){
        double val = (i+1);
        w.push_back(val);
        sum += val;
    }
    cout << "C++ attention: ";
    for(double &v : w){
        cout << v/sum << " ";
    }
    cout << endl;
    return 0;
}

File: stock_temporal_attention_multilang/novelty4_multi_head_attention/attention.py
import numpy as np

def compute_attention(data):
    # placeholder temporal attention
    weights = np.linspace(0.1, 1.0, len(data))
    weights = weights / weights.sum()
    return weights.tolist()

if __name__ == "__main__":
    data = [100,102,101,105,110]
    print("Python attention:", compute_attention(data))
